### PR TITLE
Renames the plugin to 'pytest-update-test-results'

### PR DESCRIPTION
**Pull request by LucasRochaAbraao**
Renaming the plugin so its intention is more descriptive and evident.

EDEN-2543

### CODE REVIEW

**Comment by LucasRochaAbraao** [comment]
By the way, we need to change the repository name as well.

**Comment by LucasRochaAbraao** [comment]
I also added docstrings to the modify_xml() function.

**Comment by nicoddemus** [comment]
> By the way, we need to change the repository name as well.

Done

File: src/pytest_update_test_results/update_test_results.py [hunk 1]
```diff
@@ -8,6 +8,18 @@
 def modify_xml(
     original_xml: Path, retest_results: Dict[str, TestReport], new_xml: Path
 ) -> None:
+    """Modifies an XML file containing test results by removing failed test cases that were subsequently
+    passed in a retest.
+
+        Parameters:
+            original_xml (Path): Path to the original XML file containing test results.
+            retest_results (Dict[TestReport]): A dictionary mapping test cases to their retest results.
+            new_xml (Path): The path to the new modified XML file that will be created.
```
Comment (nicoddemus) [inline src/pytest_update_test_results/update_test_results.py]:
```suggestion
            original_xml: Path to the original XML file containing test results.
            retest_results: A dictionary mapping test cases to their retest results.
            new_xml: The path to the new modified XML file that will be created.
```

No need to include the types given they are already in the signature.

Btw which docstring format is this?

**Review by nicoddemus — APPROVED** [review]

File: src/pytest_update_test_results/update_test_results.py (hunk 1 above)
Comment (LucasRochaAbraao) [reply src/pytest_update_test_results/update_test_results.py]: I just checked from the example I used, the docstring format is from [Google](https://google.github.io/styleguide/pyguide.html), but I'll check on confluence if I find the docs format we should be using at ESSS. 
And about the types, the idea was to explain what each parameter does, so I just copied the parameter list from the function definition and the types came along! I'll remove them so it's not so redundant.

Comment (nicoddemus) [reply src/pytest_update_test_results/update_test_results.py]: We use reStructuredText default

File: setup.py [hunk 2]
```diff
@@ -22,19 +22,19 @@
         "Programming Language :: Python :: 3.7",
         "Framework :: Pytest",
     ],
-    description="Pytest plugin to change XML files from a previous run.",
+    description="Pytest plugin to update the results in an XML file from a previous run.",
```
Comment (igortg) [inline setup.py]:
```suggestion
    description="Pytest plugin to update test results on a jUnit XML from previous run.",
```

File: src/pytest_update_test_results/plugin.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 from typing import Dict
 
 from _pytest.reports import TestReport
-from pytest_merge_xml.merge_xml import modify_xml
+from pytest_update_test_results.update_test_results import modify_xml
 
 
 class XMLReportMergerPlugin:
```
Comment (igortg) [inline src/pytest_update_test_results/plugin.py]:
```suggestion
class UpdateTestResultsPlugin:
```

**Review by igortg — APPROVED** [review]
I'd also rename the `--merge-xml` option to `--update-xml`. What do you think?

**Comment by igortg** [comment]
I'd also rename the `--merge-xml` option to `--update-xml`. What do you think?

File: src/pytest_update_test_results/update_test_results.py (hunk 1 above)
Comment (LucasRochaAbraao) [reply src/pytest_update_test_results/update_test_results.py]: Ok thanks, will update!

File: src/pytest_update_test_results/plugin.py (hunk 3 above)
Comment (LucasRochaAbraao) [reply src/pytest_update_test_results/plugin.py]: Yes, --update-xml will be more coherent now. I'll update along with the docstring format change.

**Pull request merged**